### ISSUE TITLE
Issue downloading thumbnails

### ISSUE THREAD

**Issue by Gyurn** [libretro/RetroArch#4724]
## Description

Changed the Directory for Thumbnails onto the SD Card of Nvidia Shield Tablet K1 Android 7.0

### Expected behavior

Thumbnails to be downloaded an extracted from the SD Card folder

### Actual behavior

Downloads take for ever, and RetroArch still tries to created a Thumbnails folder on my native storage, and nothing is actually downloaded!!  

### Steps to reproduce the bug

1. Set new Directory for thumbnails on an Android device (mine was K1)
2. Online Updater -> Thumbnails (doesn't matter which, but SNES seems to be the most difficult to download)
3. Watch the excruciatingly long download, check your SD card folder for nothing, and notice a newly created "thumbnails" folder on your native storage with nothing in it!

### Bisect Results

I'm not sure if the read/write of RetroArch is the issue, my permissions in Android say it has the capability, or if there is an issue with the pulling from the github?  SNES thumbnails for example again, took a long time to download on my PC version, but they actually showed up on the PC.  

### Version/Commit
1.4.1

- RetroArch: [version/commit]

### Environment information

- OS: 32bit Android 7.0

Hope I gave enough info!  Most issues I run into with RetroArch are mostly cosmetic, that being said "cosmetics" are now why I prefer the program!  That XMB interface is great!

P.S.  Would it be possible in future builds of RetroArch to only scrap boxart/thumbnails/screenshots for games that you have in your library?  Like instead of added all the boxarts from every game you guys have, some way to check the MD5 hashes and only pulling the files associated with what you have?  An example of this would be what the creator of Gamesome did with his front end, which I used religiously until you guys blew that out of the water with your "XMB" gui!!!

**Comment by RobLoach** [comment]
How big is your SD Card? SNES itself is ~1.5 GB. Do you have a good connection? It does take a while to download...

Alternative would be to download it manually and upload it to your SD card.

**Comment by Gyurn** [comment]
Hey, my SD Card is 128gb, ~100mbps internet speed.  Ultimately I did have to download them from my desktop and transfer it over, and they work fine.  It's just out of those 10,000 files, I only really needed like 60!

**Comment by fr500** [comment]
I think this solution is terrible for mobile devices... I like the downloading on demand we had for a while....
It's better to have no thumb for a few seconds imho than unzipping 10.000 files for 60

**Comment by RobLoach** [comment]
> downloading on demand we had for a while

What did we have? Did it do stuff with http://thumbnails.libretro.com/ or something?

**Comment by fr500** [comment]
it downloaded the thumbs when you highlighted the game.
It had it's own share of problems of course

**Comment by mangax2** [comment]
There were some needlessly huge box art images being used for thumbnails (some were well over 1000px height or width).
So, I resized all of them to be no wider than 256px and no taller than 512px -- then converted them to 256 color as well.  The quality is still great and, most importantly, it resulted in a considerable size reduction.
The MAME collection alone went from 7.18gb to 2.9gb.
I just uploaded them to DropBox in case anyone else might be interested ...
https://www.dropbox.com/sh/rigty96qbeps0mq/AAAlgb1ludrSqoFXuJ_KrFfBa?dl=0
